### ISSUE TITLE
ActionCable doesn't remove connection identifier from pubsub

### ISSUE THREAD

**Issue by tomekw** [rails/rails#26015]
Rails 5.0.0, Ruby MRI 2.3.1, Redis 3.2.1

I've one `ApiChannel` and I identify streams by `api:UUID`, I've killed mobile application (`ApiChannel` client) and I still see the entry in Redis pubsub:

```
irb(main):051:0> r = Redis.new(host: "redis", db: 1)
=> #<Redis client v3.3.1 for redis://redis:6379/1>
irb(main):052:0> puts r.pubsub("channels").select { |s| s.starts_with?("api") }
api:9e005d11-6357-4c6c-9007-5c30c86eb23b
```

This way I can't track if `ActionCable.server.broadcast("api:9e005d11-6357-4c6c-9007-5c30c86eb23b", content)` returns `0` or `1` (success).

**Comment by tomekw** [comment]
Trying to prepare script to reproduce the issue consistently...

**Comment by tomekw** [comment]
OK, so run this script on a different host

``` ruby
require "websocket-client-simple"

ws = WebSocket::Client::Simple.connect("ws://API_HOST/ws?ticket[token]=TOKEN")

ws.on :open do
  ws.send '{"command": "subscribe","identifier":"{\"channel\":\"ApiChannel\"}"}'
end

loop do
  ws.send STDIN.gets.strip
end
```

and plug off the network cable on the server - the subscription will stay for more than a minute in Redis.

Do I need to use custom pings and disconnect manually in such case?

**Comment by danielrhodes** [comment]
I think I have noticed the same thing in development, especially when a connection closed abruptly. Although I didn't go too far into this, I did a test with a forked version of ActionCable and was able to speed up the dead connection cleanup by sending a websocket ping (different from ActionCable's ping) from the server to the client.

@tomekw Is this the case in production as well? For example, in development I suggest using the redis adapter as async adapter was not working very well for me. This resolved a lot of my local dev issues.
